Compilation fails on NixOS 18.09

Just upgraded to NixOS 18.09 and wanted to add jenkinsPlugins2nix to the system packages. Compilation fails with the following output:

```
[3 of 3] Compiling Nix.JenkinsPlugins2Nix ( src/Nix/JenkinsPlugins2Nix.hs, dist/build/Nix/JenkinsPlugins2Nix.o )

src/Nix/JenkinsPlugins2Nix.hs:123:15: error:
    Not in scope: ‘Nix.mkApp’
    Perhaps you meant one of these:
      data constructor ‘Nix.NApp’ (imported from Nix.Expr),
      ‘Nix._NApp’ (imported from Nix.Expr),
      ‘Nix.nApp’ (imported from Nix.Expr)
    Neither ‘Nix.Expr’ nor ‘Nix.Pretty’ exports ‘mkApp’.
    |
123 |               Nix.mkApp (Nix.mkSym "stdenv.mkDerivation") $ Nix.mkNonRecSet
    |               ^^^^^^^^^

src/Nix/JenkinsPlugins2Nix.hs:135:18: error:
    Not in scope: ‘Nix.mkApp’
    Perhaps you meant one of these:
      data constructor ‘Nix.NApp’ (imported from Nix.Expr),
      ‘Nix._NApp’ (imported from Nix.Expr),
      ‘Nix.nApp’ (imported from Nix.Expr)
    Neither ‘Nix.Expr’ nor ‘Nix.Pretty’ exports ‘mkApp’.
    |
135 |     fetchurl p = Nix.mkApp (Nix.mkSym "fetchurl") $
    |                  ^^^^^^^^^

src/Nix/JenkinsPlugins2Nix.hs:136:38: error:
    Not in scope: ‘Nix.mkUri’
    Neither ‘Nix.Expr’ nor ‘Nix.Pretty’ exports ‘mkUri’.
    |
136 |       Nix.mkNonRecSet [ "url" Nix.$= Nix.mkUri (download_url p)
    |                                      ^^^^^^^^^

src/Nix/JenkinsPlugins2Nix.hs:141:16: error:
    Not in scope: ‘Nix.mkApp’
    Perhaps you meant one of these:
      data constructor ‘Nix.NApp’ (imported from Nix.Expr),
      ‘Nix._NApp’ (imported from Nix.Expr),
      ‘Nix.nApp’ (imported from Nix.Expr)
    Neither ‘Nix.Expr’ nor ‘Nix.Pretty’ exports ‘mkApp’.
    |
141 |     mkBody p = Nix.mkApp (Nix.mkSym "mkJenkinsPlugin") $
    |                ^^^^^^^^^
builder for '/nix/store/xf61qhh05z5km0crkd52wkbw642nh22p-jenkinsPlugins2nix-0.2.0.2.drv' failed with exit code 1
```

Looks like some incompatibility with a newer version of hnix.

--- a/default.nix
+++ b/default.nix
@@ -1,0 +1,22 @@
+{ mkDerivation, ansi-wl-pprint, attoparsec, base, bimap, bytestring
+, containers, cryptohash, hnix, http-conduit, mtl
+, optparse-applicative, stdenv, tasty-hspec, text, zip-archive
+}:
+mkDerivation {
+  pname = "jenkinsPlugins2nix";
+  version = "0.2.0.2";
+  src = ./.;
+  isLibrary = true;
+  isExecutable = true;
+  libraryHaskellDepends = [
+    ansi-wl-pprint attoparsec base bytestring containers cryptohash
+    hnix http-conduit mtl text zip-archive
+  ];
+  executableHaskellDepends = [
+    ansi-wl-pprint base bimap optparse-applicative text
+  ];
+  testHaskellDepends = [ base containers tasty-hspec text ];
+  homepage = "https://github.com/Fuuzetsu/jenkinsPlugins2nix#readme";
+  description = "Generate nix for Jenkins plugins";
+  license = stdenv.lib.licenses.bsd3;
+}

--- a/jenkinsPlugins2nix.cabal
+++ b/jenkinsPlugins2nix.cabal
@@ -19,16 +19,16 @@ library
                        Nix.JenkinsPlugins2Nix.Parser
                        Nix.JenkinsPlugins2Nix.Types
   build-depends:       base >= 4.7 && < 5
-                     , ansi-wl-pprint
-                     , attoparsec
-                     , bytestring
-                     , containers
-                     , cryptohash
-                     , hnix
-                     , http-conduit
-                     , mtl
-                     , text
-                     , zip-archive
+                     , ansi-wl-pprint >=0.6  && <0.7
+                     , attoparsec     >=0.13 && <0.14
+                     , bytestring     >=0.10 && <0.11
+                     , containers     >=0.5  && <0.6
+                     , cryptohash     >=0.11 && <0.12
+                     , hnix           >=0.5  && <0.6
+                     , http-conduit   >=2.3  && <2.4
+                     , mtl            >=2.2  && <2.3
+                     , text           >=1.2  && <1.3
+                     , zip-archive    >=0.3  && <0.4
   default-language:    Haskell2010
   ghc-options:        -Wall
 
@@ -38,9 +38,9 @@ executable jenkinsPlugins2nix
   ghc-options:         -threaded -rtsopts -with-rtsopts=-N -Wall
   build-depends:       base
                      , ansi-wl-pprint
-                     , bimap
+                     , bimap                >=0.3  && <0.4
                      , jenkinsPlugins2nix
-                     , optparse-applicative
+                     , optparse-applicative >=0.14 && <0.15
                      , text
   default-language:    Haskell2010
 
@@ -52,7 +52,7 @@ test-suite jenkinsPlugins2nix-test
   build-depends:       base
                      , containers
                      , jenkinsPlugins2nix
-                     , tasty-hspec
+                     , tasty-hspec         >=1.1 && <1.2
                      , text
   ghc-options:         -threaded -rtsopts -with-rtsopts=-N -Wall
   default-language:    Haskell2010

--- a/shell.nix
+++ b/shell.nix
@@ -1,0 +1,11 @@
+{ nixpkgs ? import <nixpkgs> {}, compiler ? "default", doBenchmark ? false }:
+let
+  inherit (nixpkgs) pkgs;
+  f = import ./default.nix;
+  haskellPackages = if compiler == "default"
+                       then pkgs.haskellPackages
+                       else pkgs.haskell.packages.${compiler};
+  variant = if doBenchmark then pkgs.haskell.lib.doBenchmark else pkgs.lib.id;
+  drv = variant (haskellPackages.callPackage f {});
+in
+  if pkgs.lib.inNixShell then drv.env else drv

--- a/src/Nix/JenkinsPlugins2Nix.hs
+++ b/src/Nix/JenkinsPlugins2Nix.hs
@@ -22,6 +22,7 @@ import qualified Data.Text.Encoding as Text
 import qualified Data.Text.IO as Text
 import qualified Network.HTTP.Simple as HTTP
 import qualified Nix.Expr as Nix
+import           Nix.Expr.Shorthands ((@@))
 import qualified Nix.JenkinsPlugins2Nix.Parser as Parser
 import           Nix.JenkinsPlugins2Nix.Types
 import qualified Nix.Pretty as Nix
@@ -113,16 +114,17 @@ mkExprsFor (Config { resolution_strategy = st, requested_plugins = ps }) = do
   return $! case eplugins of
     Left err -> Left err
     Right plugins ->
-      let args = Nix.mkParamset exprs
+      let args = Nix.mkParamset exprs False
           res = Nix.mkNonRecSet $ map formatPlugin plugins
           mkJenkinsPlugin = Nix.bindTo "mkJenkinsPlugin" $
             Nix.mkFunction (Nix.mkParamset
                               [ ("name", Nothing)
                               , ("src", Nothing)
-                              ]) $
-              Nix.mkApp (Nix.mkSym "stdenv.mkDerivation") $ Nix.mkNonRecSet
+                              ]
+                              False) $
+              Nix.mkSym "stdenv.mkDerivation" @@ Nix.mkNonRecSet
                 [ Nix.inherit [ Nix.StaticKey "name"
-                              , Nix.StaticKey "src" ]
+                              , Nix.StaticKey "src" ] Nix.nullPos
                 , "phases" Nix.$= Nix.mkStr "installPhase"
                 , "installPhase" Nix.$= Nix.mkStr "cp $src $out"
                 ]
@@ -132,16 +134,16 @@ mkExprsFor (Config { resolution_strategy = st, requested_plugins = ps }) = do
 
   where
     fetchurl :: Plugin -> Nix.NExpr
-    fetchurl p = Nix.mkApp (Nix.mkSym "fetchurl") $
-      Nix.mkNonRecSet [ "url" Nix.$= Nix.mkUri (download_url p)
+    fetchurl p = Nix.mkSym "fetchurl" @@
+      Nix.mkNonRecSet [ "url" Nix.$= Nix.mkStr (download_url p)
                       , "sha256" Nix.$= Nix.mkStr (Text.pack . show $ sha256 p)
                       ]
 
     mkBody :: Plugin -> Nix.NExpr
-    mkBody p = Nix.mkApp (Nix.mkSym "mkJenkinsPlugin") $
-      Nix.mkNonRecSet $ [ "name" Nix.$= Nix.mkStr (short_name $ manifest p)
-                        , "src" Nix.$= fetchurl p
-                        ]
+    mkBody p = Nix.mkSym "mkJenkinsPlugin" @@
+      Nix.mkNonRecSet [ "name" Nix.$= Nix.mkStr (short_name $ manifest p)
+                      , "src" Nix.$= fetchurl p
+                      ]
 
     formatPlugin :: Plugin -> Nix.Binding Nix.NExpr
     formatPlugin p = short_name (manifest p) Nix.$= mkBody p

